picard/util/__init__.py

```diff
@@ -655,9 +656,14 @@ def extract_year_from_date(dt):
     """ Extracts year from  passed in date either dict or string """
 
     if isinstance(dt, Mapping):
-        year = int(dt.get('year'))
+        try:
+            return int(dt.get('year'))
+        except (TypeError, ValueError) as e:
+            log.debug(e)
```

Comment:
I wonder if we really need to log this here. It's kind of expected functionality here that it might fail, and the function reacts to this by returning None. IMHO a `pass` or `return None` would be ok here and below.

Reply:
Done in 4f592a8